Compilation failure with Qt 6

These lines of codes failed to build with Qt 6.2.4 with Core5Compat
The errors I got are:

(1)
https://github.com/f4exb/sdrangel/blob/54288cf4c70e33e0588b3e3f8e6bb68b2b41de7b/sdrbase/maincore.cpp#L361-L362
```
sdrangel/sdrbase/maincore.cpp:361:60: error: ‘updateTimeout’ is not a member of ‘QGeoPositionInfoSource’
  361 |         connect(m_positionSource, &QGeoPositionInfoSource::updateTimeout, this, &MainCore::positionUpdateTimeout);
      |                                                            ^~~~~~~~~~~~~
sdrangel/sdrbase/maincore.cpp:362:106: error: no match for call to ‘(const QOverload<QGeoPositionInfoSource::Error>) (QGeoPositionInfoSource::Error (QGeoPositionInfoSource::*)() const)’
  362 |         connect(m_positionSource, qOverload<QGeoPositionInfoSource::Error>(&QGeoPositionInfoSource::error), this, &MainCore::positionError);
      |           
```
(2)
https://github.com/f4exb/sdrangel/blob/54288cf4c70e33e0588b3e3f8e6bb68b2b41de7b/sdrgui/mainwindow.cpp#L177-L180
'''
sdrangel/sdrgui/mainwindow.cpp:177:15: error: ‘class QScreen’ has no member named ‘setOrientationUpdateMask’
  177 |     screen()->setOrientationUpdateMask(Qt::PortraitOrientation
      |               ^~~~~~~~~~~~~~~~~~~~~~~~
'''
(3)
https://github.com/f4exb/sdrangel/blob/54288cf4c70e33e0588b3e3f8e6bb68b2b41de7b/sdrgui/gui/dialogpositioner.cpp#L32-L35

'''
sdrangel/sdrgui/gui/dialogpositioner.cpp:32:23: error: ‘class QScreen’ has no member named ‘setOrientationUpdateMask’
   32 |     dialog->screen()->setOrientationUpdateMask(Qt::PortraitOrientation
'''
(4)
https://github.com/f4exb/sdrangel/blob/54288cf4c70e33e0588b3e3f8e6bb68b2b41de7b/plugins/channelrx/demodft8/ft8demodsettingsdialog.h#L24
'''
(5)
sdrangel/build/plugins/channelrx/demodft8/demodft8_autogen/EWIEGA46WW/../../../../../../plugins/channelrx/demodft8/ft8demodsettingsdialog.h:24:7: error: using typedef-name ‘using QStringList = class QList<QString>’ after ‘class’
   24 | class QStringList;
      |       ^~~~~~~~~~~
'''
(6)
https://github.com/f4exb/sdrangel/blob/54288cf4c70e33e0588b3e3f8e6bb68b2b41de7b/plugins/channelrx/heatmap/heatmapgui.cpp#L221
'''
sdrangel/plugins/channelrx/heatmap/heatmapgui.cpp:221:82: error: no matching function for call to ‘max(int, qsizetype)’
  221 |     ui->averagePeriod->setMinimum(std::max(1, m_averagePeriodTexts.size() - value));
      | 
'''
(7)
https://github.com/f4exb/sdrangel/blob/54288cf4c70e33e0588b3e3f8e6bb68b2b41de7b/plugins/channelrx/heatmap/heatmapgui.cpp#L699
'''
sdrangel/plugins/channelrx/heatmap/heatmapgui.cpp:699:82: error: no matching function for call to ‘max(int, qsizetype)’
  699 |     ui->averagePeriod->setMinimum(std::max(1, m_averagePeriodTexts.size() - value));
      |  
'''

Hi! Is it to be expected that I am getting the exact first error @Mohamedadlyi also got in this issue?
Unfortunately, I am lost on how to debug or fix it.

First of all, as requested in the wiki, due to lack of knowledge I can't show any evidence of something being wrong in the code. I am commenting only because it is the same error. Feel free to disregard this comment if this is not sufficient (really, I don't want to add noise or drain development time).

QT version: 6.4.3 (platform is ArchLinux)
qt6-5compat is also installed (on ArchLinux, there are no 'core' packages regarding qt)
sdrangel commit: d67aac3b277080fa697b8a0eaa47c0de5f5c0597 (current master, as of 2023-03-20T21:20+00:00)
The build procedure has been as instructed from the wiki
Here is the error. Hope this can help:
```
[ 15%] Building CXX object sdrbase/CMakeFiles/sdrbase.dir/maincore.cpp.o
/home/damiano/compile/sdrangel/build/sdrangel/sdrbase/maincore.cpp: In member function ‘void MainCore::initPosition()’:
/home/damiano/compile/sdrangel/build/sdrangel/sdrbase/maincore.cpp:364:75: error: no match for call to ‘(const QOverload<QGeoPositionInfoSource::Error>) (QGeoPositionInfoSource::Error (QGeoPositionInfoSource::*)() const)’
  364 |         connect(m_positionSource, qOverload<QGeoPositionInfoSource::Error>(&QGeoPositionInfoSource::error), this, &MainCore::positionError);
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<details>
<summary>Full log of the error</summary>

```
[ 15%] Building CXX object sdrbase/CMakeFiles/sdrbase.dir/maincore.cpp.o
/home/damiano/compile/sdrangel/build/sdrangel/sdrbase/maincore.cpp: In member function ‘void MainCore::initPosition()’:
/home/damiano/compile/sdrangel/build/sdrangel/sdrbase/maincore.cpp:364:75: error: no match for call to ‘(const QOverload<QGeoPositionInfoSource::Error>) (QGeoPositionInfoSource::Error (QGeoPositionInfoSource::*)() const)’
  364 |         connect(m_positionSource, qOverload<QGeoPositionInfoSource::Error>(&QGeoPositionInfoSource::error), this, &MainCore::positionError);
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/qt6/QtCore/qglobalstatic.h:4,
                 from /usr/include/qt6/QtCore/QGlobalStatic:1,
                 from /home/damiano/compile/sdrangel/build/sdrangel/sdrbase/maincore.cpp:18:
/usr/include/qt6/QtCore/qglobal.h:1320:20: note: candidate: ‘template<class R, class T> constexpr decltype (ptr) QConstOverload<Args>::operator()(R (T::*)(Args ...) const) const [with T = R; Args = {QGeoPositionInfoSource::Error}]’
 1320 |     constexpr auto operator()(R (T::*ptr)(Args...) const) const noexcept -> decltype(ptr)
      |                    ^~~~~~~~
/usr/include/qt6/QtCore/qglobal.h:1320:20: note:   template argument deduction/substitution failed:
/home/damiano/compile/sdrangel/build/sdrangel/sdrbase/maincore.cpp:364:75: note:   candidate expects 2 arguments, 1 provided
  364 |         connect(m_positionSource, qOverload<QGeoPositionInfoSource::Error>(&QGeoPositionInfoSource::error), this, &MainCore::positionError);
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/qt6/QtCore/qglobal.h:1308:20: note: candidate: ‘template<class R, class T> constexpr decltype (ptr) QNonConstOverload<Args>::operator()(R (T::*)(Args ...)) const [with T = R; Args = {QGeoPositionInfoSource::Error}]’
 1308 |     constexpr auto operator()(R (T::*ptr)(Args...)) const noexcept -> decltype(ptr)
      |                    ^~~~~~~~
/usr/include/qt6/QtCore/qglobal.h:1308:20: note:   template argument deduction/substitution failed:
/home/damiano/compile/sdrangel/build/sdrangel/sdrbase/maincore.cpp:364:75: note:   types ‘R (T::)(QGeoPositionInfoSource::Error)’ and ‘QGeoPositionInfoSource::Error (QGeoPositionInfoSource::)() const’ have incompatible cv-qualifiers
  364 |         connect(m_positionSource, qOverload<QGeoPositionInfoSource::Error>(&QGeoPositionInfoSource::error), this, &MainCore::positionError);
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/qt6/QtCore/qglobal.h:1337:20: note: candidate: ‘template<class R> constexpr decltype (ptr) QOverload<Args>::operator()(R (*)(Args ...)) const [with Args = {QGeoPositionInfoSource::Error}]’
 1337 |     constexpr auto operator()(R (*ptr)(Args...)) const noexcept -> decltype(ptr)
      |                    ^~~~~~~~
/usr/include/qt6/QtCore/qglobal.h:1337:20: note:   template argument deduction/substitution failed:
/home/damiano/compile/sdrangel/build/sdrangel/sdrbase/maincore.cpp:364:75: note:   mismatched types ‘R (*)(QGeoPositionInfoSource::Error)’ and ‘QGeoPositionInfoSource::Error (QGeoPositionInfoSource::*)() const’
  364 |         connect(m_positionSource, qOverload<QGeoPositionInfoSource::Error>(&QGeoPositionInfoSource::error), this, &MainCore::positionError);
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-inconsistent-missing-override’ may have been intended to silence earlier diagnostics
```
</details>

Given this is the same error as above, before opening a new issue I would like to be sure this is an actual problem, and not one just on my part.
I am open to hints on which might the problem be and how could this be solved (though, since I am inexperienced in c++ and I don't know qt, I would be of little help, I am afraid)

--- a/plugins/channelrx/demodnavtex/navtexdemodsink.cpp
+++ b/plugins/channelrx/demodnavtex/navtexdemodsink.cpp
@@ -355,7 +355,7 @@ void NavtexDemodSink::receiveBit(bool bit)
                         getMessageQueueToChannel()->push(msg);
                     }
                     // Add character to message buffer
-                    m_messageBuffer.append(c);
+                    m_messageBuffer.append(QChar(c));
                 }
                 else
                 {

--- a/plugins/channelrx/heatmap/heatmapgui.cpp
+++ b/plugins/channelrx/heatmap/heatmapgui.cpp
@@ -695,7 +695,7 @@ void HeatMapGUI::displaySettings()
 
     value = (int)std::log10(m_settings.m_sampleRate);
     ui->sampleRate->setValue(value);
-    int idx = std::min(std::max(0, value-2), m_sampleRateTexts.size() - 1);
+    int idx = std::min(std::max(0, value-2), (int)m_sampleRateTexts.size() - 1);
     ui->sampleRateText->setText(m_sampleRateTexts[idx]);
     ui->averagePeriod->setMinimum(std::max(1, static_cast<int> (m_averagePeriodTexts.size()) - value));
 

--- a/sdrbase/maincore.cpp
+++ b/sdrbase/maincore.cpp
@@ -358,10 +358,12 @@ void MainCore::initPosition()
     if (m_positionSource)
     {
         connect(m_positionSource, &QGeoPositionInfoSource::positionUpdated, this, &MainCore::positionUpdated);
-        #if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
+#if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
         connect(m_positionSource, &QGeoPositionInfoSource::updateTimeout, this, &MainCore::positionUpdateTimeout);
-        #endif
         connect(m_positionSource, qOverload<QGeoPositionInfoSource::Error>(&QGeoPositionInfoSource::error), this, &MainCore::positionError);
+#else
+        connect(m_positionSource, &QGeoPositionInfoSource::errorOccurred, this, &MainCore::positionError);
+#endif
         m_position = m_positionSource->lastKnownPosition();
         m_positionSource->setUpdateInterval(1000);
         m_positionSource->startUpdates();

